Sources/BreakpointManager.cpp

```diff
@@ -162,11 +162,6 @@ bool BreakpointManager::hit(Address const &address) {
 
   it->second.type =
       static_cast<Type>(it->second.type & ~kTypeTemporaryUntilHit);
-  if (!it->second.type) {
```

Comment:
Removing this is good. Can you just add a small comment before line 163 to mention that the breakpoint will be `erase`'d when we call `BreakpointManager::disable()` if its `type` field goes to 0?